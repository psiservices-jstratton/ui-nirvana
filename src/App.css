.App {
  text-align: center;
}

.App-header {
  background-color: #cccccc;
  min-height: 10vh;
  display: flex;
  flex-direction: column;
  align-items: left;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}
