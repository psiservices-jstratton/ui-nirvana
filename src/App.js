import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import './App.css';
import Nav from './components/navigation/NavBar';
import Loading from './components/Loading';
import Home from './pages/Home';
const TestDriver = React.lazy(() => import('./pages/TestDriver'));

function App() {
    return (
        <div className="App">
            <React.Suspense fallback={<Loading />}>
                <Router>
                    <Nav />
                    <Switch>
                        <Route exact path="/">
                            <Home />
                        </Route>
                        <Route path="/test">
                            <TestDriver />
                        </Route>
                    </Switch>
                </Router>
            </React.Suspense>
        </div>
    );
}

export default App;
