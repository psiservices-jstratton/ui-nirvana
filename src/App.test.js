import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, waitForElement } from '@testing-library/react';
import App from './App';


describe('<App />',  () => {
    // Considered a basic 'smoke test'
    it('renders without crashing', () => {
        const AppRoot = render(<App />);
        expect(AppRoot).toMatchSnapshot();
    });

    it('should begin on home page', async () => {
        const history = createMemoryHistory();
        const { getByText } = render(
            <Router history={history}>
                <App />
            </Router>
        );
        // Stupid assertion. Probably a better way to check a home page header or that we're on the initial
        // app route?
        const homeHeadingNode = await waitForElement(() => getByText('This is the home page.'));
        expect(homeHeadingNode).toHaveTextContent('This is the home page.');
    });
});
