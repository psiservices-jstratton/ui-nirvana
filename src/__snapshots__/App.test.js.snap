// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav
          class="psi-nav-bar navbar navbar-expand-md navbar-light bg-light"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            <img
              alt="Psi logo"
              src="logo.png"
            />
          </a>
          <ul
            class="nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link psi-nav-link"
                href="/test"
              >
                Test
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-auto"
            >
              <h2
                role="heading"
              >
                This is the home page.
              </h2>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav
        class="psi-nav-bar navbar navbar-expand-md navbar-light bg-light"
      >
        <a
          class="navbar-brand"
          href="/"
        >
          <img
            alt="Psi logo"
            src="logo.png"
          />
        </a>
        <ul
          class="nav"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link psi-nav-link"
              href="/test"
            >
              Test
            </a>
          </li>
        </ul>
      </nav>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col-auto"
          >
            <h2
              role="heading"
            >
              This is the home page.
            </h2>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
